htdocs/fourn/class/fournisseur.product.class.php

```diff
@@ -383,6 +383,12 @@ public function update_buyprice($qty, $buyprice, $user, $price_base_type, $fourn
 				if (empty($error))
 				{
 					$this->db->commit();
+					if (!empty($conf->dynamicprices->enabled) && !empty($this->fk_price_expression)) {
+						// Computed selling prices may depend on the purchase price. By re-fetching the product, we
+						// force it to re-compute the selling price and to save it if it has changed.
+						$product = new Product($this->db);
+						if ($product->fetch($this->id) > 0) $product->updatePriceFromExpression();
```

Comment:
Can you confirm you need to create a new object Product.
Why not just $this->updatePriceFromExpression()

Reply:
Sorry, this was the remnant of an older code layout which required it, but it is no longer necessary. I modified the code accordingly.